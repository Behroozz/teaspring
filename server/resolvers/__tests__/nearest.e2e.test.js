const gql = require('graphql-tag')
const mongoose = require('mongoose')
const { server } = require('../../app')
const { startTestServer, toPromise } = require('../../helpers/testUtils')

const POST_ANSWER = gql`
  mutation postingAnswer {
	postAnswers(scenario_id: "ade831c4-dd44-4205-b21b-1ac5304ebf75") {
		scenario_id
    answers {
      inks 
    }
  }
}
`

describe('Server - e2e', () => {
  let stop, graphql

  beforeAll(async () => {
    await mongoose.connect('mongodb://localhost/teaspring', { useNewUrlParser: true, useCreateIndex: true }, (err) => {
      if (err) {
        console.error(err)
        process.exit(1)
      }
    })
  })

  beforeEach(async () => {
    const testServer = await startTestServer(server)
    stop = testServer.stop
    graphql = testServer.graphql
  })

  afterEach(() => stop())

  it('Get list of answers', async () => {
    const res = await toPromise(
      graphql({
        query: POST_ANSWER,
        variables: {},
      }),
    )

    expect(res).toMatchSnapshot()
    expect(1).toEqual(1)
  })
})
