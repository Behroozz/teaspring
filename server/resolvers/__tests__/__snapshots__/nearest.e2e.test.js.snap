// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e Get list of answers 1`] = `
Object {
  "data": Object {
    "postAnswers": Object {
      "answers": Array [
        Object {
          "inks": Array [
            "HX0453",
            "AN0123",
          ],
        },
        Object {
          "inks": Array [
            "PJ4273",
          ],
        },
      ],
      "scenario_id": "ade831c4-dd44-4205-b21b-1ac5304ebf75",
    },
  },
}
`;
